draft-ietf-scitt-architecture.md

```diff
@@ -551,64 +578,131 @@ Editor's Note: This may be moved to appendix.
-  - Error code `AwaitingDIDResolutionTryLater`
-  - Error code `InvalidInput`
+- Status 201 - Registration is successful.
+  - Header `Location: <Base URL>/entries/<Entry ID>`
+  - Header `Content-Type: application/json`
+  - Body `{ "entryId": "<Entry ID"> }`
 
-[TODO] Use 5xx for AwaitingDIDResolutionTryLater
+- Status 202 - Registration is pending.
+  - Header `Location: <Base URL>/operations/<Operation ID>`
+  - Header `Content-Type: application/json`
+  - (Optional) Header: `Retry-After: <seconds>`
+  - Body `{ "operationId": "<Operation ID>", "status": "pending" }`
 
-The `201` response contains the `x-ms-ccf-transaction-id` HTTP header which can be used to retrieve the Registration Receipt with the given transaction ID. [TODO] this has to be made generic
+- Status 400 - Registration was unsuccessful due to invalid input.
+  - Error code `badSignatureAlgorithm`
+  - Error code `TBD`
 
-[TODO] probably a bad idea to define a new header, or is it ok? can we register a new one? https://www.iana.org/assignments/http-fields/http-fields.xhtml
+If 202 is returned, then clients should wait until registration succeeded or failed by polling the registration status using the Operation ID returned in the response. Clients should always obtain a receipt as a proof that registration has succeeded.
 
-The `400` response has a `Content-Type: application/json` header and a body containing details about the error:
+### Retrieve Operation Status
 
-```json
-{
-  "error": {
-    "code": "<error code>",
-    "message": "<message>"
-  }
-}
-```
+#### Request
+
+~~~
+GET <Base URL>/operations/<Operation ID>
+~~~
 
-`AwaitingDIDResolutionTryLater` means the service does not have an up-to-date DID document of the DID referenced in the Signed Claims but is performing or will perform a DID resolution after which the client may retry the request. The response may contain the HTTP header `Retry-After` to inform the client about the expected wait time.
+#### Response
+
+One of the following:
 
-`InvalidInput` means either the Signed Claims message is syntactically malformed, violates the signing profile (e.g. signing algorithm), or has an invalid signature relative to the currently resolved DID document.
+- Status 200 - Registration is pending
+    - Header: `Content-Type: application/json`
+    - (Optional) Header: `Retry-After: <seconds>`
+    - Body: `{ "operationId": "<Operation ID>", "status": "pending" }`
 
-### Retrieve Registration Receipt
+- Status 200 - Registration was successful
+    - Header: `Location: <Base URL>/entries/<Entry ID>`
+    - Header: `Content-Type: application/json`
+    - Body: `{ "operationId": "<Operation ID>", "status": "succeeded", "entryId": "<Entry ID>" }`
+
+- Status 200 - Registration failed
+    - Header `Content-Type: application/json`
+    - Body: `{ "operationId": "<Operation ID>", "status": "failed", "error": { "type": "<type>", "detail": "<detail>" } }`
+    - Error code: `badSignatureAlgorithm`
+    - Error code: `TBD`
```

Comment:
looks like a leftover?

Reply:
Not really, but I made it more clear: "- [TODO]: more error codes to be defined"